patches/minecraft/net/minecraft/advancements/PlayerAdvancements.java.patch

```diff
@@ -0,0 +1,33 @@
+--- ../src-base/minecraft/net/minecraft/advancements/PlayerAdvancements.java
++++ ../src-work/minecraft/net/minecraft/advancements/PlayerAdvancements.java
+@@ -27,6 +27,8 @@
+ import net.minecraft.util.JsonUtils;
+ import net.minecraft.util.ResourceLocation;
+ import net.minecraft.util.text.TextComponentTranslation;
++import net.minecraftforge.common.MinecraftForge;
++import net.minecraftforge.event.entity.player.AdvancementEvent;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
+ 
+@@ -208,12 +210,21 @@
+ 
+             if (!flag1 && advancementprogress.func_192105_a())
+             {
++                if (MinecraftForge.EVENT_BUS.post(new AdvancementEvent.Pre(this.field_192762_j, p_192750_1_))) {
++                    advancementprogress.func_192101_b(p_192750_2_);
++                    this.func_193764_b(p_192750_1_);
++                    this.field_192761_i.remove(p_192750_1_);
++                    return false;
++                }
++
+                 p_192750_1_.func_192072_d().func_192113_a(this.field_192762_j);
+ 
+                 if (p_192750_1_.func_192068_c() != null && p_192750_1_.func_192068_c().func_193220_i() && this.field_192762_j.field_70170_p.func_82736_K().func_82766_b("announceAdvancements"))
+                 {
+                     this.field_192756_d.func_184103_al().func_148539_a(new TextComponentTranslation("chat.type.advancement." + p_192750_1_.func_192068_c().func_192291_d().func_192307_a(), new Object[] {this.field_192762_j.func_145748_c_(), p_192750_1_.func_193123_j()}));
+                 }
++
++                MinecraftForge.EVENT_BUS.post(new AdvancementEvent.Post(this.field_192762_j, p_192750_1_));
```

Comment:
...And change this line to `net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.entity.player.AdvancementEvent.Post(this.field_192762_j, p_192750_1_));`